### PR TITLE
Fix undefined behaviour of the aaa analysis

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Rizin report warning messages increasingly when repeated run aaa.
This is due to the fact that analysis information aren't reset between
2 aaa analysis so the blocks analysis isn't deterministic.

The simple 'hack' is to force the fcns analysis data to be reset.

I add a simple test to verify that the second analysis isn't overriding
the first one.

I believe that if this ´solution´ is accepted, we should update the aaa help.

And the test suite will fail, because I didn´t add the binary to https://github.com/rizinorg/rizin-testbins. I would like to have some advice on this first.

**Test plan**

see  #329 and the new test

**Closing issues**

 #329

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -4771,6 +4771,10 @@ static bool isSkippable(RzBinSymbol *s) {
 	return false;
 }
 
+static bool has_old_analysis_value(RZ_BORROW RZ_NONNULL RzCore *core) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: I am not sure `RZ_BORROW/RZ_NONNULL` here are necessary, it's very short and static function.

**Comment by ghost** [comment]
@XVilka Do you think it is an acceptable solution/patch?

**Comment by XVilka** [comment]
I don't think it's a right fix, but not 100% sure.

**Comment by ret2libc** [comment]
What about functions defined by the user? I'm not sure either this is the right fix...

**Comment by ghost** [comment]
~The bug come from the type matching analysis for all functions (aaft).~

rz_analysis_dwarf_integrate_functions introduce some side effect.

It introduce some side effect that impact (aa).

**Comment by ghost** [comment]
see #945, I switch to a better src branch. Sorry.

**Pull request closed without merging**